dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1108,13 +1108,14 @@ private void provisionWorkflowVersionUploadFiles(ApiClient zendoClient, Deposit
         // Add workflow version source files as a zip to the DOI upload deposit
         // Borrow code from getWorkflowZip
         Set<SourceFile> sourceFiles = workflowVersion.getSourceFiles();
-        java.nio.file.Path path = Paths.get(workflowVersion.getWorkingDirectory());
+        java.nio.file.Path path = Paths.get(System.getProperty("java.io.tmpdir"));
```

Comment:
I suggest calling  `Files.createTempDirectory()`, then deleting the directory. If you write all files to the same directory, you have the possibility of them overwriting each other, since you don't give them unique names.

Reply:
The bigger issue is that getWorkingDirectory is user provided which is dangerous

Reply:
done and refactored code